ceph-dev-pipeline/build/Jenkinsfile

```diff
@@ -120,6 +154,73 @@ pipeline {
               }
               sh "sha256sum dist/*"
               sh "cat dist/sha1 dist/version"
+              script {
+                def props = readProperties file: "${WORKSPACE}/dist/other_envvars"
+                for (p in props) {
+                  env."${p.key}" = p.value
+                }
+              }
+              sh '''#!/bin/bash
+                set -ex
+                cd dist
+                mkdir ceph
+                tar --strip-components=1 -C ceph -xjf ceph-$VERSION.tar.bz2
+              '''
+            }
+          }
+          stage("check for built packages") {
+            when {
+              environment name: 'THROWAWAY', value: 'false'
+              environment name: 'CI_COMPILE', value: 'true'
+            }
+            environment {
+              CHACRACTL_KEY = credentials('chacractl-key')
+              SHAMAN_API_KEY = credentials('shaman-api-key')
+            }
+            steps {
+              script {
+                sh './scripts/setup_chacractl.sh'
+                def chacra_url = sh(
+                  script: '''grep url ~/.chacractl | cut -d'"' -f2''',
+                  returnStdout: true,
+                ).trim()
+                def os = get_os_info(env.DIST)
+                def chacra_endpoint = "ceph/${env.BRANCH}/${env.SHA1}/${os.name}/${os.version_name}/${env.ARCH}/flavors/${env.FLAVOR}/"
+                def chacractl_rc = sh(
+                  script: "$WORKSPACE/.venv/bin/chacractl exists binaries/${chacra_endpoint}",
+                  returnStatus: true,
+                )
+                if ( chacractl_rc == 0 && env.FORCE != "true" ) {
+                  println("Skipping compilation since chacra already has artifacts. To override, use THROWAWAY=true (to skip this check) or FORCE=true (to re-upload artifacts).")
+                  env.CI_COMPILE = "false"
+                }
+              }
+            }
+          }
+          stage("builder container") {
```

Comment:
"builder container" hits me as a bit odd grammatically, but it's no big deal.

Reply:
yeah, I chose that phrasing to avoid confusion over using "build" as a noun vs. a verb. this is shorter than "assemble container image that will eventually compile" ;)